8thwall-camera.js

```diff
@@ -5,88 +5,402 @@
+
+    getPermissions: async function () {
+        // iOS "feature". If we want to request the DeviceMotion permission, user has to interact with the page at first (touch it).
+        // If there was no interaction done so far, we will render a HTML overlay with would get the user to interact with the screen
+
+        if (DeviceMotionEvent && DeviceMotionEvent.requestPermission) {
+            try {
+                const result = await DeviceMotionEvent.requestPermission();
+
+                // The user must have rejected the motion event on previous page load. (safari remembers this choice).
+                if (result !== "granted") {
+                    throw new Error("MotionEvent");
+                }
+            } catch (exception) {
 
-        const vals = ['auto', 'back', 'front'];
-        this.camera = vals[this.camera];
-        if(this.camera == 'auto') {
-            this.camera = 'back';
+                // User had no interaction with the page so far
+                if (exception.name === "NotAllowedError") {
+                    const motionEvent = await this.promptForDeviceMotion();
+                    if (motionEvent !== "granted") {
+                        throw new Error("MotionEvent");
+                    }
+                } else {
+                    throw new Error("MotionEvent");
+                }
+            }
+        }
+
+        try {
+            // make sure we get the camera stream
+            const stream = await navigator.mediaDevices.getUserMedia({ video: true, audio: true })
+
+            // If we successfully acquired the camera stream - we can stop it and wait until 8thwall requests it again
+            stream.getTracks().forEach((track) => {
+                track.stop();
+            });
+
+        } catch (exception) {
+            throw new Error("Camera");
         }
+    },
+
 
-        this.onStart = this.onStart.bind(this);
+    init: function () {
+        this.view = this.object.getComponent("view");
         this.onUpdate = this.onUpdate.bind(this);
+        this.onAttach = this.onAttach.bind(this);
+        this.onException = this.onException.bind(this);
+        this.onCameraStatusChange = this.onCameraStatusChange.bind(this);
+    },
+
+    start: async function () {
+        console.log("useCustomUIOverlays", this.useCustomUIOverlays);
+        if (!this.useCustomUIOverlays) {
+            console.log("Initing")
+            OverlaysHandler.init();
+        }
 
+        try {
+            await this.getPermissions();
+        } catch (error) {
+            // User did not grant the camera or motionEvent permissions
+            window.dispatchEvent(new CustomEvent("8thwall-permission-fail", {detail: error}))
+            return;
+        }
+
+        await waitForXR8();
+
+        XR8.XrController.configure({
+            // enableLighting: true,
+            disableWorldTracking: false
+        });
+
+        this.GlTextureRenderer = XR8.GlTextureRenderer.pipelineModule();
```

Comment:
```suggestion
        this.glTextureRenderer = XR8.GlTextureRenderer.pipelineModule();
```

Minor code-style nit-pick